### PR TITLE
DAG which runs standalone import-app

### PR DESCRIPTION
**Pull request by RoelWKramer**
DAG with podoperator python task

AB#49052

### CODE REVIEW

File: dags/benk/nap/dag.py [hunk 1]
```diff
@@ -7,60 +8,53 @@
 team_name = "BenK"
 workload_name = "NAP"
 dag_id = team_name + "_" + workload_name
-# TODO: Figure out in which environment this dag is running,
-#  or set an env var with the container_image url in the airflow UI.
-container_image = "benkontacr.azurecr.io/benk-airflow-task:development"
-command = ["/bin/sh", "-c", "/app/entrypoint.sh"]
+
```
Comment (RoelWKramer) [inline dags/benk/nap/dag.py:13]: This works, but needs cleanup.

File: dags/benk/nap/dag.py [hunk 2]
```diff
@@ -7,60 +8,53 @@
...
-        hostnetwork=True,
-        get_logs=True,
-        in_cluster=True,
-        dag=dag,
-        log_events_on_failure=True
+        log_events_on_failure=True,
+        # image_pull_secrets=[V1LocalObjectReference('regcred')],
+        # configmaps=["benks-map"],
+        # Configure the TEST var in the admin to make this DAG work.
+        env_vars={"TEST_ENV_VAR": "{{ var.value.TEST }}"},
```
Comment (RoelWKramer) [inline dags/benk/nap/dag.py]: python -c "import os; print(os.environ)" shows "TEST_ENV_VAR". Environment works.

File: dags/benk/nap/common.py [hunk 3]
```diff
@@ -1,15 +1,46 @@
...
 
 default_args = {
     "owner": "basis en kernregistraties",
     "depends_on_past": False,
     "start_date": datetime.utcnow(),
-    "email": [
-        "roel.kramer@amsterdam.nl",
-        "roel.de.vries@amsterdam.nl"
-    ],
+    "email": ["gob.ois@amsterdam.nl"],
```
Comment (roel-devries) [inline dags/benk/nap/common.py]:
```suggestion
    "email": ["ois.gob@amsterdam.nl"],
```

File: dags/benk/environment.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            if _is_env_var(k)
+        ]
+
+
+class GrondslagEnvironment(OperatorEnvironment):
+    """Settings and secrets to connect to the grondslag database."""
+
+    GRONDSLAG_DATABASE_HOST = Variable.get(
+        "GRONDSLAG_DATABASE_HOST", default_var="DBEXTA1LORA2.ACC.AMSTERDAM.NL"
+    )
```
Comment (roel-devries) [inline dags/benk/environment.py]: hide default credentials

File: dags/benk/environment.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+class OperatorEnvironment:
+    """Base environment class to put environment variables on.
+
+    TODO: Use 'Secret' object for passwords instead.
+    """
+    def env_vars(self) -> List[V1EnvVar]:
+        """Returns all env vars in this object."""
+
+        def _is_env_var(k) -> bool:
```
Comment (roel-devries) [inline dags/benk/environment.py:17]: Maybe a warning in case of invalid env var?

Comment (RoelWKramer) [reply dags/benk/environment.py:17]: nah; that would cause a lot of warnings. Maybe just check if it is a Variable instead.

Comment (roel-devries) [reply dags/benk/environment.py:17]: Thats even better indeed, if we only use Variables there

Comment (RoelWKramer) [reply dags/benk/environment.py:17]: impossible though; as the values are returned at class creation time, not object creation time.

**Review by roel-devries — APPROVED** [review]

**Pull request merged**